### DISCUSSION TITLE
Upstreaming quality changes (e.g. MISRA)

### DISCUSSION THREAD

**Discussion by craigmoulder** [OpenAMP/open-amp#300] [General]
Hi,

I work for NXP, and for quality reasons we are considering doing a static analysis sweep (e.g. MISRA) of the open-amp and libmetal code.

This effort would ultimately touch a large amount of the code and introduce considerable churn.  Based on past experiences, we would do this in a pragmatic way - fix violations it makes sense to fix, and deviate those that it doesn't (e.g. things that would sacrifice code readability).

What is the general opinion/consensus on upstreaming improvements like this?  I want to get a sense of whether or not this kind of high-impact change would be welcomed, and if we should undertake the effort with upstreaming in mind.

Thanks,

Craig

**Comment by arnopo** [comment]
Hi Craig,

> Hi,
> 
> I work for NXP, and for quality reasons we are considering doing a static analysis sweep (e.g. MISRA) of the open-amp and libmetal code.

Yes, this is a topic we have discussed in the past, we are interested. Personally, I run [c-stat](https://www.iar.com/cstat), which, I hope,  should have already some of the the MISRA issues.

> 
> This effort would ultimately touch a large amount of the code and introduce considerable churn. Based on past experiences, we would do this in a pragmatic way - fix violations it makes sense to fix, and deviate those that it doesn't (e.g. things that would sacrifice code readability).

That seems to me very reasonable.   Perhaps a first step to move forward it to have a result of a MISRA run to have a better view of the potential work to do to be MISRA compliant?

> 
> What is the general opinion/consensus on upstreaming improvements like this? I want to get a sense of whether or not this kind of high-impact change would be welcomed, and if we should undertake the effort with upstreaming in mind.

Without a picture of the issue detected it is quite difficult to answer. Improving the code robustness using tools such as MISRA seems to me important. As you mention just need to be pragmatic.  At least the critical violations should be fixed. 

As a first step, could you enter an Issue with the list of points detected by your MISRA tools?

> 
> Thanks,
> 
> Craig

Thanks for you proposal!
Arnaud
